alua.c

```diff
@@ -560,17 +561,55 @@ int alua_implicit_transition(struct tcmu_device *dev, struct tcmulib_cmd *cmd)
 		return ret;
 
 	pthread_mutex_lock(&rdev->state_lock);
-	if (rdev->lock_state == TCMUR_DEV_LOCK_LOCKED) {
+	if (rdev->lock_state == TCMUR_DEV_LOCK_WRITE_LOCKED) {
+		/* For both read/write cases in this state is good */
 		goto done;
-	} else if (rdev->lock_state == TCMUR_DEV_LOCK_LOCKING) {
-		tcmu_dev_dbg(dev, "Lock acquisition operation is already in process.\n");
+	} else if (rdev->lock_state == TCMUR_DEV_LOCK_WRITE_LOCKING) {
+		/* For both read/write cases in this state should return busy */
+		tcmu_dev_dbg(dev, "Write lock acquisition operation is already in process.\n");
 		ret = TCMU_STS_BUSY;
 		goto done;
+	} else if (is_read && !rdev->lock_lost) {
+		if (rdev->lock_state == TCMUR_DEV_LOCK_READ_LOCKED) {
+			goto done;
+		} else if (rdev->lock_state == TCMUR_DEV_LOCK_READ_LOCKING) {
```

Comment:
Nit: can this be bumped up above line 572 and then you could delete the extra check on lines 605-609?

Reply:
Yeah, makes sense, I will do it.
Thanks.

Reply:
Nit: so now this statement will not be reached so it can be removed

Reply:
Done.